besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1404,6 +1404,16 @@ private void validateGoQuorumCompatibilityModeParam() {
         throw new IllegalStateException(
             "GoQuorum compatibility mode (enabled) can only be used if genesis file has 'isQuorum' flag set to true.");
       }
+      genesisConfigOptions
+          .getChainId()
```

Comment:
what needs to happen if it is not present?

Reply:
if you don't specify chainId in genesis, it uses network if provided (cmd line) and if not provided then it's mainnet

Reply:
the validation against chainId is in a different place to network (because we don't read genesis file until later) but it might make sense to to them both later, together, to make it easier to read

Reply:
refactored into a single method